### ISSUE TITLE
[Bug] Incorrect base being set for PWA plugin

### ISSUE THREAD

**Issue by piyush-das** [Mister-Hope/vuepress-theme-hope#475]
Hi, 
I am using the @mr-hope/vuepress-plugin-pwa, with the following configuration.
```js
const pwaOptions = {
  cachePic : true,
  base : '/testdocs/',
  favicon : '/img/favicon.ico',
};

module.exports = {
  title: "xyz",
  description: 'xyz',
  head: head(pwaOptions,[ 
    ['link', { rel: 'icon', href: '/img/favicon.ico' }],
    ['meta', { name: 'theme-color', content: '#11a8cd'}], 
  ]),
  base: '/testdocs/',
...
...
```

Since the site is being hosted as a subfolder. The base path for the vuepress is set as `base: '/testdocs/'`. However, it seems that setting the `base` configuration in the pwa configuration causes multiple failures.

- /manifest.webmanifest:1 Failed to load resource: the server responded with a status of 404 () [ The `base` doesn't seem to be added to the path.]
- 'https://xyz.gitlab.io/testdocs/testdocs/manifest.webmanifest' [ Some errors relating to base being replicated in the url for manifest]

Please let me know if I am missing or doing some wrong configuration.

Thanks

**Comment by Mister-Hope** [comment]
Gotcha, I have a really important exam at dec 26-27, so I will leave it and fix it after the exam.

By the way, are you using just the plugin? With theme it seems to work well.

**Comment by Mister-Hope** [comment]
https://github.com/Mister-Hope/vuepress-theme-hope/blob/4170746e6aa0b816e2676b0301c4a11442a6cfe6/packages/pwa/node/head.ts#L52

I did not find bugs at a glance.

**Comment by Mister-Hope** [comment]
Bug in dev server has been found and has been fix.

**Comment by Mister-Hope** [comment]
base option is removed and everything should work fine in 1.7.3

**Comment by piyush-das** [comment]
@Mister-Hope Thanks, this seems to have the correct base settings. In my current setup, I have hosted the vuepress docs on gitlab pages. Since the project is a private project. For accessing the documentation the gitlab authentication needs to be done. However, after logging in to gitlab, the service workers seem to be working fine but manifest is not being fetched. I am getting the following errors in the console.
![Screenshot from 2020-12-10 18-13-51](https://user-images.githubusercontent.com/68941107/101773988-84997900-3b13-11eb-8a45-1f65ade997fb.png)
![Screenshot from 2020-12-10 18-14-27](https://user-images.githubusercontent.com/68941107/101774068-9b3fd000-3b13-11eb-9073-ea3e6008889c.png)

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://projects.gitlab.io/auth?domain=https://aigenresearch.gitlab.io&state=KlyvJ_5Q1Z1No7QuaA==. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://projects.gitlab.io/auth?domain=https://aigenresearch.gitlab.io&state=KlyvJ_5Q1Z1No7QuaA==. (Reason: CORS request did not succeed).
```

And the network tab shows the following : 
![Screenshot from 2020-12-10 18-11-02](https://user-images.githubusercontent.com/68941107/101773725-2f5d6780-3b13-11eb-9edc-c698b3f4be0f.png)

I am not sure whether this is due to an error in the pwa plugin or because of gitlab hosting. Any help would be greatly appreciated.

Thanks

**Comment by Mister-Hope** [comment]
I am not familiar with gitlab. But everything should be fetched in the same site in my plugin since I am not accessing domain name at all.

If you are metting cross site issues, it's probably gitlab's issue.

**Comment by piyush-das** [comment]
Ok Thanks

**Comment by piyush-das** [comment]
@Mister-Hope I looked into the issue and found that the issue was due to `crossOrigin : "use-credentials"` missing the in the `link` for `manifest`. The solution I referenced on [stackoverflow](https://stackoverflow.com/a/52170318). I guess due to the added authentication introduced due to gitlab, credentials are required for accessing resources.
Updating line 37 in head.js, solves the issue.
```js
setLink("manifest", "/manifest.webmanifest", {crossOrigin:"use-credentials"});
```
Do you plan to incorporate this feature or should I use a locally maintained version of your great plugin for the same.

Thanks,

**Comment by Mister-Hope** [comment]
I checked out the information you provided, and I did not see any possible side effects adding `crossorigin` attribute.

I will add it later.

**Comment by piyush-das** [comment]
@Mister-Hope. It seems that the `base` is still not propagating everywhere. [PWAInstallModal](https://github.com/Mister-Hope/vuepress-theme-hope/blob/5b99eebd33b78588c9fbb634ae234b6dc6fefe2b/packages/pwa/client/PWAInstallModal.ts#L77)  requires the base path to be included. Additionally, any of the icons specified within `pwa` `manifest` option still require the paths to include the `base` manually.

Thanks

**Comment by Mister-Hope** [comment]
The fetch link in PWAInstallModal is a bug.

The manifest options in my opinion, is the final manifest content in js format.

**Comment by Mister-Hope** [comment]
I prefer not to handle or change any value in it, while I will provide some fallback for some propertys. Whether handle the links or not is both confusing.

User can store a manifest in the public folder. And users may want to use links from the root url. A subfolder site having the same icons as the root site is quite common. Handling links in the manifest options can be confused to users if they just want to access the url with actual root base.

**Comment by Mister-Hope** [comment]
Oh by the way, I think it's possible to set releative url in `manifest` option to avoid adding base.

manifest crossorigin and PWAInstallModal is fixed in 1.7.7.

**Comment by piyush-das** [comment]
On a side note. I have tried setting `maxPicSize : 1230`. Since I have an illustration gif which is around 1.1 MB. However still during build I get the following 
```
Warnings: Skipped gifs/xyz.gif, as its 1067 KB.
```
Is there something else which needs to be done to include this gif ?

Thanks

**Comment by Mister-Hope** [comment]
Emm, my fault, I am using picMaxSize in code

**Comment by Mister-Hope** [comment]
Fixed in 1.7.8

**Comment by piyush-das** [comment]
Works great now. Should I close the issue now?

Thanks again for the great work.
